subprojects/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegritySpec.groovy

```diff
@@ -47,11 +52,17 @@ class DistributionIntegritySpec extends DistributionIntegrationSpec {
 
         when:
         def jarsWithDuplicateFiles = [:]
+        def classesIndex = [:] as HashMap<String, List<String>> // class name -> list of containing jars
         jars.each { jar ->
             new ZipFile(jar).withCloseable {
                 def names = it.entries()*.name
                 def groupedNames = names.groupBy { it }
                 groupedNames.each { name, all ->
+                    if (name.endsWith("class") && !name.endsWith("module-info.class") && !name.endsWith("package-info.class")) {
```

Comment:
It's a corner case, but this would also accept things named `not_a_class`.

Reply:
`.class` would be better.  I'll update.